modules/apps/29-fee/module.go

```diff
@@ -191,7 +201,23 @@ func (am AppModule) OnChanOpenInit(
 	counterparty channeltypes.Counterparty,
 	version string,
 ) error {
-	return nil
+	feeVersion, appVersion := channeltypes.SplitChannelVersion(version)
+	if feeVersion != types.Version {
+		return sdkerrors.Wrapf(types.ErrInvalidVersion, "expected: %s, got: %s", types.Version, feeVersion)
+	}
+	// Claim channel capability passed back by IBC module
+	if err := am.scopedKeeper.ClaimCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID)); err != nil {
```

Comment:
Question: Will every middleware application want to do this claim capability step? Is the idea that the first middleware will always create the capability and just pass it along to the next?

Reply:
If the middleware is acting on information coming from app to IBC, then this is a useful pattern to enforce that communication from app to IBC must go through middleware first.

This is a good design pattern to enforce anyway since it enforces the middleware setup in code.

Reply:
Gotcha, thanks!